Sources/SwiftParser/Statements.swift

```diff
@@ -352,6 +357,30 @@ extension Parser {
       fatalError("Macros are not allowed in this position!")
     }
   }
+  
+  /// Parse a `#_hasSymbol` condition.
+  ///
+  /// Grammar
+  /// =======
+  ///
+  ///     has-symbol-condition → '#_hasSymbol' '(' expr ')'
+  @_spi(RawSyntax)
+  public mutating func parsePoundHasSymbolConditionElement() -> RawSyntax {
+    assert(self.at(any: [.poundHasSymbolKeyword]))
+    let keyword = self.consumeAnyToken()
```

Comment:
Could you change this to 
```suggestion
    let (unexpectedBeforeKeyword, keyword) = self.expect(.poundHasSymbolKeyword)
```

Should we want to support recovery in front of `#_hasSymbol`, that will allow us to do so more easily by changing
```diff
- if self.at(.poundHasSymbolKeyword) {
+ if self.canRecoverTo(.poundHasSymbolKeyword) != nil {
    return self.parsePoundHasSymbolConditionElement()
  }
```

Not saying that we should support recovery in this case but it’s good practice to prepare for that path.

Reply:
Sure! I followed the patterns I found in the parsing for `if #available` - should that also be updated?